File: Array/caiweixiong/c/homework_017.c
//
// Created by Vashon on 2020/6/24.
//

#include <stdio.h>

// 统计位数为偶数的数字，LeetCode第1295题
void homework_017_1295(void) {
    int findNumbers(int *nums, int numsSize);
    int nums[] = {12, 345, 2, 6, 7896};
    printf("the double bit num have : %d", findNumbers(nums, 5));
}

int findNumbers(int *nums, int numsSize) {
    if (!nums || numsSize <= 0)
        return 0;
    int count = 0, bit = 0, temp = 0;
    for (int i = 0; i < numsSize; ++i) {
        bit = 0;
        temp = nums[i];
        do {
            temp /= 10;
            bit++;
        } while (temp != 0);
        if ((bit % 2) == 0)
            count++;
    }
    return count;
}

File: Array/caiweixiong/java/Homework_017.java
public class Homework_017 {

    // 统计位数为偶数的数字，LeetCode第1295题
    void homework_017_1295() {
        int[] nums = {12, 345, 2, 6, 7896};
        System.out.println(String.format("the double bit num have : %d", findNumbers(nums)));
    }

    private int findNumbers(int[] nums) {
        if (nums == null || nums.length <= 0) {
            return 0;
        }
        int count = 0, bit = 0, value = 0;
        for (int i = 0; i < nums.length; i++) {
            bit = 0;
            value = nums[i];
            do {
                value /= 10;
                bit++;
            } while (value != 0);
            if ((bit % 2) == 0)
                count++;
        }
        return count;
    }
}
